File: Day0903/app/src/main/java/com/example/day0903/MainActivity.java
package com.example.day0903;

import android.content.Intent;
import android.os.Bundle;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    ActivityResultLauncher<Intent> launcher;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });


        launcher = registerForActivityResult(
                new ActivityResultContracts.StartActivityForResult(),
                result -> {
                    if (result.getResultCode() == RESULT_OK) {
                        Intent intent = result.getData();
                        if (intent != null) {
                            String replace = intent.getStringExtra("replace");
                            EditText edName = findViewById(R.id.edName);
                            edName.setText(replace);
                        }
                    }
                }
        );


        findViewById(R.id.btnSend).setOnClickListener(v -> {
            Intent intent = new Intent(this, MainActivity2.class);
            EditText edName = findViewById(R.id.edName);
            intent.putExtra("name", edName.getText().toString());
            launcher.launch(intent);
        });
    }
}

File: Day13_01/app/src/main/java/com/example/day13_01/MainActivity.kt
package com.example.day13_01

import android.content.Intent
import android.os.Bundle
import android.widget.*
import androidx.activity.enableEdgeToEdge
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private lateinit var mBtnGoSub: Button
    private lateinit var mTvMsg: TextView
    private lateinit var mEdMsg: EditText
    private lateinit var mSubActivityResultLauncher: ActivityResultLauncher<Intent>


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        mBtnGoSub = findViewById(R.id.btn_go_sub)
        mTvMsg = findViewById(R.id.tvMsg)
        mEdMsg = findViewById(R.id.edMsg)


        mSubActivityResultLauncher = registerForActivityResult(
            ActivityResultContracts.StartActivityForResult()) { result ->
            if (result.resultCode == RESULT_OK) {
                val _data = result.data
                val _msg = _data?.getStringExtra("msg")
                mTvMsg.text = _msg
            }
        }

        mBtnGoSub.setOnClickListener {
            val _inputIntent = Intent(this, SubActivity::class.java)
            _inputIntent.putExtra("msg", mEdMsg.text.toString())
            mSubActivityResultLauncher.launch(_inputIntent)
        }
    }
}